std_execution.bs

```diff
@@ -3022,3 +3039,152 @@ are all well-formed.
         does not call `std::terminate` if an error occurs after control is returned to the caller, the program is ill-formed with no diagnostic required.
 
     2. Otherwise, `execution::start_detached(execution::then(execution::schedule(sch), f))`.
+
+## Coroutine utilities <b>[execution.coro_utils]</b> ## {#spec-execution.coro_utils}
+
+### `execution::as_awaitable` <b>[execution.coro_utils.as_awaitable]</b> ### {#spec-execution.coro_utils.as_awaitable}
+
+1. `as_awaitable` is used to wrap a sender into an object that is awaitable within a particular coroutine. A sender `s` of type `S` can be made awaitable if and only if:
+
+    - `S` models the concept `typed_sender`, and
+
+    - It passes zero or one values of a unique type to `execution::set_value`, and
+
+    - It passes only objects of type `exception_ptr` to `exception::set_error`, and
+
+    - The promise type of the awaiting coroutine implements a function called `unhandled_done` that takes no arguments and returns a `coroutine_handle<>`.
+
+    <pre highlight="c++">
+    // exposition only
+    template&lt;class S>
+      using <i>single-sender-value-type</i> = <i>see below</i>;
+
+    // exposition only
+    template&lt;class S>
+      concept <i>single-typed-sender</i> =
+        typed_sender&lt;S> &&
+        requires { typename <i>single-sender-value-type</i>&lt;S>; };
+
+    template&lt;<i>single-typed-sender</i> Sender, class Promise>
+      <i>unspecified</i> as_awaitable(Sender&& s, Promise& p);
+    </pre>
+
+2. Alias template <i>single-sender-value-type</i> is defined as follows:
+
+    - If `sender_traits<remove_cvref_t<S>>::value_types<Tuple, Variant>` would have the form `Variant<Tuple<T>>`, then <code><i>single-sender-value-type</i>&lt;S></code> is an alias for type `T`.
+
+    - Otherwise, if `sender_traits<remove_cvref_t<S>>::value_types<Tuple, Variant>` would have the form `Variant<Tuple<>>` or `Variant<>`, then <code><i>single-sender-value-type</i>&lt;S></code> is an alias for type `void`.
```

Comment:
I wonder if the result is `Variant<>` whether we should specify that the `await_resume()` method of the returned awaitable will be tagged as `[[noreturn]]`?

Reply:
I'm not going to sweat this for now.